src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -252,17 +253,19 @@ private void DispatchInvocation(InvocationMessage invocation, CancellationToken
             handler.Handler(invocation.Arguments);
         }
 
-        private void DispatchInvocationStreamItem(StreamItemMessage streamItem, InvocationRequest irq)
+        // This async void is GROSS but we need to dispatch asynchronously because we're writing to a Channel
+        // and there's nobody to actually wait for us to finish.
+        private async void DispatchInvocationStreamItemAsync(StreamItemMessage streamItem, InvocationRequest irq)
         {
             _logger.LogTrace("Received StreamItem for Invocation #{invocationId}", streamItem.InvocationId);
 
             if (irq.CancellationToken.IsCancellationRequested)
             {
                 _logger.LogTrace("Canceling dispatch of StreamItem message for Invocation {invocationId}. The invocation was cancelled.", irq.InvocationId);
             }
-            else
+            else if (!await irq.StreamItem(streamItem.Item))
```

Comment:
Only in the Streaming case, if someone else closes the write end of the channel (which only we control). I'll add some guard code though since this runs in a background thread.